ee/clickhouse/queries/funnels/funnel_correlation_persons.py

```diff
@@ -1,144 +1,189 @@
             else ""
         )
 
-        event_join_query = self._funnel_correlation._get_events_join_query()
+        actor_join_subquery, actor_join_subquery_params = self._get_actor_subquery()
+        group_filters, group_filters_params = self._get_group_filters()
 
         query = f"""
             WITH
                 funnel_people as ({funnel_persons_query}),
-                toDateTime(%(date_to)s) AS date_to,
-                toDateTime(%(date_from)s) AS date_from,
-                %(target_step)s AS target_step,
-                %(funnel_step_names)s as funnel_step_names
+                %(target_step)s AS target_step
             SELECT
-                DISTINCT person.person_id as person_id
-            FROM events AS event
-                {event_join_query}
-                AND event.event = %(target_event)s
-                {conversion_filter}
-                {prop_query}
+                DISTINCT funnel_people.person_id as person_id
+            FROM funnel_people
+            {actor_join_subquery}
+            WHERE {conversion_filter}
+            {group_filters}
             ORDER BY person_id
             LIMIT {self._filter.correlation_person_limit}
             OFFSET {self._filter.correlation_person_offset}
         """
-
         params = {
             **funnel_persons_params,
-            **prop_params,
-            "target_event": self._filter.correlation_person_entity.id,
-            "funnel_step_names": [entity.id for entity in self._filter.events],
+            **actor_join_subquery_params,
+            **group_filters_params,
             "target_step": len(self._filter.entities),
         }
 
         return query, params
 
-    def _format_results(self, results: List[Tuple[str]]):
-        people = Person.objects.filter(team_id=self._team.pk, uuid__in=[val[0] for val in results])
+    def _get_actor_subquery(self) -> Tuple[str, Dict[str, Any]]:
+        if self.is_aggregating_by_groups:
+            actor_join_subquery, actor_join_subquery_params = GroupsJoinQuery(
+                self._filter, self._team.pk, join_key="funnel_people.person_id"
+            ).get_join_query()
+        else:
+            person_query, actor_join_subquery_params = ClickhousePersonQuery(
+                self._filter,
+                self._team.pk,
+                entity=Entity(
+                    {"id": "person", "type": "events", "properties": self._filter.correlation_property_values}
+                ),
+            ).get_query()
+
+            actor_join_subquery = f"""
+                JOIN ({person_query}) person
+                ON person.id = funnel_people.person_id
+            """
 
-        from posthog.api.person import PersonSerializer
+        return actor_join_subquery, actor_join_subquery_params
 
-        return (
-            PersonSerializer(people, many=True).data,
-            len(results) > cast(int, self._filter.correlation_person_limit) - 1,
-        )
+    def _get_group_filters(self):
```

Comment:
did this instead of morphing the group filtering into the subquery directly. Open to other options